File: util/Animable.java
package common.util;

import common.io.json.JsonClass;
import common.io.json.JsonField;
import common.pack.Source.ResourceLocation;
import common.util.anim.AnimI;
import common.util.anim.EAnimI;

@JsonClass
public abstract class Animable<A extends AnimI<A, T>, T extends Enum<T> & AnimI.AnimType<A, T>> extends ImgCore {

	@JsonField(alias = ResourceLocation.class)
	public A anim;

	public abstract EAnimI getEAnim(T t);

}

File: util/Animable.kt
package common.util

import common.util.anim.AnimI
import common.util.anim.AnimI.AnimType
import common.util.anim.EAnimI

abstract class Animable<A : AnimI<A, T>?, T> : ImgCore() where T : Enum<T>?, T : AnimType<A, T>? {
    var anim: A? = null
    abstract fun getEAnim(t: T): EAnimI?
}
